src/physics_common.cpp

```diff
@@ -23,4 +24,155 @@ void russian_roulette(Particle* p)
   }
 }
 
+//==============================================================================
+// IMPORTANCE_SPLIT
+//==============================================================================
+
+void split_by_ratio(Particle* p, double importance_ratio) {
+  p->wgt_last_ = p->wgt_;
+  double importance_rounded = std::floor(importance_ratio);
+  
+  int num_splits = 0;
+  // how many splits - probabilistic 3/2 split 
+  //  half the time we get 3 and half we get 2
+  if ( prn() < importance_ratio - importance_rounded )
+    num_splits = int(importance_rounded);
+  else 
+    num_splits = int(std::ceil(importance_ratio));
+  // calculate the tru number of splits
+  p->wgt_ /= float(num_splits);
+  // note 1 -> n not 0->n
+  for ( int i = 1 ; i < num_splits ; i++ ) {
+    p->create_secondary(p->u(),p->E_,p->type_);
+  }
+
+  return;
+}
+
+//==============================================================================
+// ROULETTE
+//==============================================================================
+
+void roulette(Particle* p, double importance_ratio) {
+  // this is a kill event
+  double random = prn();
+  //std::cout << random << " " << importance_ratio;
+  if ( random >= importance_ratio) {
+    //std::cout << " kill" << std::endl;
+    p->alive_ = false;
+    p->wgt_ = 0.;
+    p->wgt_last_ = 0.;
+  } else {
+    //std::cout << " survive " << std::endl;
+    p->wgt_last_ = p->wgt_;
+    //std::cout << p->last_wgt << " ";
+    p->wgt_ /= importance_ratio;
+    //std::cout << p->wgt << std::endl;
+  }
+  return;
+}
```

Comment:
Could this replace the existing `russian_roulette`?

Reply:
This replaces parts of `russian_roulette`